api/v1alpha1/inferencepool_types.go

```diff
@@ -59,8 +59,85 @@ type InferencePoolSpec struct {
 	// +kubebuilder:validation:Maximum=65535
 	// +kubebuilder:validation:Required
 	TargetPortNumber int32 `json:"targetPortNumber"`
+
+	// EndpointPickerConfig configures the extension that runs the endpoint picking service.
+	// to this pool.
+	EndpointPickerConfig `json:"endpointPickerConfig"`
+}
+
+type EndpointPickerConfig struct {
+	// Extension configures an endpoint picker as an extension service.
+	//
+	// +optional
+	Extension *ExtensionConfig `json:"extension"`
+}
+
+// ExtensionConfig specifies how to configure an extension that runs the endpoint picker.
+type ExtensionConfig struct {
+	// ExtensionRef is a reference to a service extension.
+	ExtensionRef *ExtensionReference `json:"extensionRef"`
```

Comment:
- Include `// +optional` since this field is a pointer.
- Include `,omitempty` in the json tag for optional fields.